node_modules/oae-preview-processor/lib/api.js

```diff
@@ -507,18 +507,18 @@ var _generatePreviews = function(ctx, callback) {
         if (err) {
             return callback(err);
         } else if (processor) {
-            processor.generatePreviews(ctx, ctx.content, function(err, ignored) {
+            processor.generatePreviews(ctx, ctx.content, function(err, status) {
                 if (err) {
-                    ctx.setStatus('error', callback);
-                } else if (ignored) {
-                    ctx.setStatus('ignored', callback);
+                    ctx.setStatus(ContentConstants.previews.ERROR, callback);
+                } else if (status && _.contains(ContentConstants.previews, status)) {
+                    ctx.setStatus(status, callback);
                 } else {
                     ctx.savePreviews(callback);
                 }
             });
         } else {
             log().info({'contentId': ctx.contentId, 'content': ctx.content, 'revision': ctx.revision}, 'Ignoring as this type of content is not supported for now');
-            ctx.setStatus('ignored', callback);
+            ctx.setStatus(ContentConstants.previews.IGNORED, callback);
         }
```

Comment:
A bit of a nitpick, but this recommendation is in the spirit of avoiding creating asynchronous branches inside if statements:

We have 2 "escape" procedures here where we don't want to just log / promote something then break right away: the error case and the "ignored" case. Then we have our happy path which is to run the generate previews function. We typically want to do the escape routes first: stop processing immediately and promote the error condition. Then our happy path can live outside the escape conditions (i.e., not within the if statement) in the main body of the function. So it might look more like this:

```javascript
if (err) {
  return callback(err);
} else if (!processor) {
  log().info({'contentId': ctx.contentId, 'content': ctx.content, 'revision': ctx.revision}, 'Ignoring as this type of content is not supported for now');
  return ctx.setStatus(ContentConstants.previews.IGNORED, callback);
}

processor.generatePreviews(ctx, ctx.content, function(err, status) {
  if (err) {
    return ctx.setStatus('error', callback);
  } else if (ignored) {
    return ctx.setStatus(ContentConstants.previews.IGNORED, callback);
  } else if (status && _.contains(ContentConstants.previews, status)) {
    return ctx.setStatus(status, callback);
  }

  return ctx.savePreviews(callback);
});
```